### ISSUE TITLE
Internal server error on admin UI when external model is not available

### ISSUE THREAD

**Issue by guystreeter** [matiasdelellis/facerecognition#571]
If the external model is configured, but the app cannot connect to it, the facerecognition setting admin page displays an internal server error instead of a useful message.

To recreate the problem, shutdown the external model, or set the URL to something else, then select the Face Recognition tab Administration settings.

In the nextcloud log:
```
{"reqId":"L80aah93SABTB3o65Eyj","level":3,"time":"2022-04-16T17:29:30+00:00","remoteAddr":"127.0.0.1","user":"poot","app":"index","method":"GET","url":"/settings/admin/facerecognition","message":"Cannot connect to external model: Failed to connect to predator.local port 5000: Connection refused","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36","version":"23.0.3.2","exception":{"Exception":"Exception","Message":"Cannot connect to external model: Failed to connect to predator.local port 5000: Connection refused","Code":0,"Trace":[{"file":"/var/www/html/custom_apps/facerecognition/lib/Model/ExternalModel/ExternalModel.php","line":107,"function":"open","class":"OCA\\FaceRecognition\\Model\\ExternalModel\\ExternalModel","type":"->","args":[]},{"file":"/var/www/html/custom_apps/facerecognition/lib/Settings/Admin.php","line":80,"function":"getMaximumArea","class":"OCA\\FaceRecognition\\Model\\ExternalModel\\ExternalModel","type":"->","args":[]},{"file":"/var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php","line":129,"function":"getForm","class":"OCA\\FaceRecognition\\Settings\\Admin","type":"->","args":[]},{"file":"/var/www/html/apps/settings/lib/Controller/AdminSettingsController.php","line":83,"function":"formatSettings","class":"OCA\\Settings\\Controller\\AdminSettingsController","type":"->","args":[{"20":[{"__class__":"OCA\\FaceRecognition\\Settings\\Admin","modelManager":{"__class__":"OCA\\FaceRecognition\\Model\\ModelManager"},"settingsService":{"__class__":"OCA\\FaceRecognition\\Service\\SettingsService"},"memoryLimits":{"__class__":"OCA\\FaceRecognition\\Helper\\MemoryLimits"}}]}]},{"file":"/var/www/html/apps/settings/lib/Controller/CommonSettingsTrait.php","line":140,"function":"getSettings","class":"OCA\\Settings\\Controller\\AdminSettingsController","type":"->","args":["facerecognition"]},{"file":"/var/www/html/apps/settings/lib/Controller/AdminSettingsController.php","line":68,"function":"getIndexResponse","class":"OCA\\Settings\\Controller\\AdminSettingsController","type":"->","args":["admin","facerecognition"]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":217,"function":"index","class":"OCA\\Settings\\Controller\\AdminSettingsController","type":"->","args":["facerecognition"]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":126,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Settings\\Controller\\AdminSettingsController"},"index"]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Settings\\Controller\\AdminSettingsController"},"index"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Settings\\Controller\\AdminSettingsController","index",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"section":"facerecognition","_route":"settings.AdminSettings.index"}]},{"file":"/var/www/html/lib/base.php","line":1006,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/settings/admin/facerecognition"]},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/custom_apps/facerecognition/lib/Model/ExternalModel/ExternalModel.php","Line":145,"CustomMessage":"--"}}
```

**Comment by matiasdelellis** [comment]
These are the things that I have to review.. Thank you!

**Comment by WolfsWelpe** [comment]
Hey I just encountered this error after a fresh install on Ubuntu 22.04 Jammy. I have used the easy way for installation:
https://github.com/matiasdelellis/facerecognition/wiki/PDlib-Installation with pdlib and installed the app over the nextcloud app store.
I followed this guide to install Nextcloud but without the ssl configuration because my nextcloud instance is running behind a reverse proxy. https://www.c-rieger.de/nextcloud-installationsanleitung-apache2/

What should I do?
